### PR TITLE
feat: add Try HCP callout to /<product> landing

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds a "Try HCP" callout, specifically `TryHcpCalloutPrebuilt`, to `/<product>` landing views that have an HCP offering.

## 🛠️ How

- Adds `hasHcpCalloutContent` check function, to determine whether a given product slug has an HCP callout to render
- Adds `TryHcpCalloutPrebuilt` to product landing views, behind a `hasHcpCalloutContent(product.slug)` conditional

## 📸 Design Screenshots

> **Note**: `/vagrant` and `/nomad` product landing views do not have any changes, as they don't have an HCP offering.

### Design spec

🎨 [Figma link][figma]

CleanShot 2022-11-22 at 11 00 43@2x

### Screenshots from this PR

| Product | Screenshot |
| - | - |
| HCP | hcp |
| Terraform | terraform | 
| Packer | packer |
| Consul | consul | 
| Vault | vault |
| Boundary | boundary |
| Waypoint | waypoint |

## 🧪 Testing

- [ ] Visit a product landing view that has an HCP-ish offering
    - These product landing pages are: [/hcp][/hcp], [/terraform][/terraform], [/packer][/packer], [/consul][/consul], [/vault][/vault], [/boundary][/boundary], and [/waypoint][/waypoint].
    - An HCP callout should appear in the sidecar, on desktop viewports
    - On smaller viewports, the sidecar and the HCP callout within it should not be visible
- [ ] Visit an install view that does _not_ have an HCP-ish offering
    - These product landing pages are:  [/vagrant][/vagrant] and [/nomad][/nomad]
    - The sidecar should appear the same as upstream, without an HCP callout

### Note on authoring mechanism

The content for these callouts is currently authored within our `src/components/try-hcp-callout` component folder. We could move the callout content to be authorable in `src/content/<product>/product-landing.json`, similar to the "get started card" content.

However, given we need to maintain pre-built content for MDX contexts, it felt like using that pre-built content on landing pages was a reasonable path forward. If we want the content for these callouts to diverge from the same callouts used in MDX contexts, that's something we can very feasibly do once the need arises.

[task]: https://app.asana.com/0/0/1203365375583140/f
[preview]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app
[figma]: https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=15357%3A196108&t=Uk5ZEStcihqUIkse-4
[/hcp]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/hcp
[/terraform]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/terraform
[/packer]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/packer
[/consul]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/consul
[/vault]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/vault
[/boundary]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/boundary
[/waypoint]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/waypoint
[/nomad]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/nomad
[/vagrant]: https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/vagrant

### CODE REVIEW

**Comment by emilypersson1** [comment]
@zchsh Looking good so far - only issue I see is in Consul, it says "Try HCP for free" instead of "Try HCP Consul for free" https://dev-portal-git-zstry-cloud-on-landing-pages-hashicorp.vercel.app/consul

**Review by emilypersson1 — APPROVED** [review]
Looks great, thanks for that update @zchsh! FYI Kelly is out this week so I believe @heitzke will be taking a look at this PR

File: src/views/product-landing/product-landing.module.css [hunk 1]
```diff
@@ -11,7 +11,11 @@
 }
 
 .getStartedMargin {
-	margin: 48px 0;
+	margin: 48px 0 32px 0;
+}
+
+.tryHcpCalloutMargin {
```
Comment (kendallstrautman) [inline src/views/product-landing/product-landing.module.css:17]: The designs don't seem to show a consistent margin for this component, so I would just check in about that if you haven't already.

**Review by kendallstrautman — APPROVED** [review]
👏🏻 🚀

File: src/views/product-landing/product-landing.module.css (hunk 1 above)
Comment (zchsh) [reply src/views/product-landing/product-landing.module.css:17]: Good call - @heitzke tagging ya for review here, let me know if the margin around the component looks alright! 🙇 💖  (screenshots are in the description of this PR)

**Pull request merged**